src/OrchardCore.Modules/OrchardCore.Taxonomies/Fields/TaxonomyField.cs

```diff
@@ -7,5 +7,11 @@ public class TaxonomyField : ContentField
     {
         public string TaxonomyContentItemId { get; set; }
         public string[] TermContentItemIds { get; set; } = Array.Empty<string>();
+
+        /// <summary>
+        /// Only populated when using the tag editor, and contains the tag display text.
+        /// Requires resaving the content item if updating the taxonomy tag term display text.
+        /// </summary>
+        public string[] Tags { get; set; } = Array.Empty<string>();
```

Comment:
It shouldn't be explicit in this class. The `Elements` property on the field should be used instead.

Reply:
That would only be acceptable if it was more like `TermNames`, and not specific to only tags. But it can also be handled by an extension method on this field, tp make it nicer that exposing the dynamic properties.

Reply:
Agree, and done. Have used the `Content` property rather than `Elements`, as Tags would have had to become it's own `TagElement : ContentItem` class.
Extension method used for `field.GetTags()` (not `SetTags`), seemed unnecessary), and the liquid content accessor just continues to handle them as Content.Tags

Reply:
Moved this to the elements array by creating a `TagElement` which inherits from `ContentElement` (with an extension method for easier access)

This makes the field accessor look like `Tags.TagElement.Tags`
i.e. `FieldName.TagElement.Tags`

We should decide if we like that pattern for semi known properties on a field. 
This way it is a little more protected against name conflicts, as we have specified a class that holds the name of the element.